### ISSUE TITLE
[Feature] A couple challenge features

### ISSUE THREAD

**Issue by shrublet** [Drewol/unnamed-sdvx-clone#417]
Was trying to set up some challenge courses to replicate the SDVX Skill Analysis courses from the main games and thought of a couple ideas that may be useful for people making challenge courses if you hadn't already planned to implement them yourself.

**Permissive guage**
Foremost, it'd be great if permissive gauge were added as an option for challenges. The exact rates of gauge depletion are on this chart and also detailed on the [JP Bemani wiki](https://bemaniwiki.com/index.php?SOUND%20VOLTEX%20VIVID%20WAVE/%A5%B7%A5%B9%A5%C6%A5%E0#RATE_STYLE), but I couldn't find the exact depletion rates when below 30%. Though technically permissive includes hp carrying over through tracks, that's already that feature in challenges and not sure if that inherently _needs_ to be tied to permissive gauge as it can also be utilized as a generally less punishing gauge if not tied to carry over.
![image](https://user-images.githubusercontent.com/63182228/103513140-d8e7fe00-4e1e-11eb-940f-3c747611e9d9.png)

**Sets**
In the arcade games, for each dan, there is multiple courses which contribute to an overall average achievement rate which is just an average of the all of the courses achievement rates within each set. I think it would be a valuable addition to USC, especially as how they're being used for the KSM Discord Skill Analyzer sets with multiple challenges each with their own percentage requirements within their own categories similar to arcade Skill Analyzer. I've also attached a pic below of how it looks in-game for reference.
![image](https://user-images.githubusercontent.com/63182228/103517782-8a8b2d00-4e27-11eb-9278-2f5b3d44ddcc.png)

**Comment by Drewol** [comment]
Hi, thanks for the requests but I would appreciate it if you made separate issues for each feature so that they can be closed individually when a feature gets completed or rejected (though I don't see much reason to reject either of these)

**Comment by shrublet** [comment]
Apologies- wasn’t sure if it was more cumbersome to open multiples issues, but didn’t think about it in that regard. Would you like me to open another issue and separate them?

**Comment by Drewol** [comment]
Yeah that'd be better, you can probably just copy the description of each feature you wrote here.
